aiida/manage/configuration/options.py

```diff
@@ -30,6 +32,14 @@
         'description': 'The polling interval in seconds to be used by process runners',
         'global_only': False,
     },
+    'daemon.default_workers': {
+        'key': 'daemon_default_workers',
+        'valid_type': 'int',
+        'valid_values': None,
+        'default': DEFAULT_DAEMON_WORKERS,
+        'description': 'The default number of workers to be launched by `verdi daemon start`',
```

Comment:
Just to understand - e.g. if I use `incr` does this value change?
In which sense is it a default rather than the actual number of daemon workers?

Reply:
`verdi daemon incr` is an independent command. It still takes an argument to tell how many workers should be added, `verdi daemon incr N` will add `N` workers with default being 1. The new option only applies how many workers are started automatically when invoking `verdi daemon start`. Already you can do `verdi daemon start 10` to start 10 workers straight away, but now you can configure it to have it start 10 by default. Saves an entire three key strokes ;)

Reply:
Ok, I just wanted to make sure.
I think some users might assume that if you do `verdi daemon incr 1` and then restart the daemon that you would keep the increased number of workers rather than resetting to this default.

It might make sense to  mention explicitly in the docstring of `verdi daemon incr` that these changes are reset whenever you stop the daemon, and that you should use this new config option to make permanent changes.

Reply:
It is a bit more subtle than that even. Doing `verdi daemon restart` will restart the workers but not the daemon itself, i.e. you will keep how every many workers that were running before the restart command. Doing `verdi daemon restart --reset` literally does `verdi daemon stop && verdi daemon start` so there it will start whatever is the default for `verdi daemon start`, which can now be configured through this new option

Reply:
This was already explained in the docstring of `verdi daemon restart`:
```
    By default will only reset the workers of the running daemon. After the restart the same amount of workers will be
    running. If the `--reset` flag is passed, however, the full circus daemon will be stopped and restarted with just
    a single worker.
```
I just noticed that we just have to update the last bit with respect to the new default behavior.

Reply:
I have updated the docstrings of `verdi daemon start` and `verdi daemon restart` in the first commit to reflect the new behavior and give more detail